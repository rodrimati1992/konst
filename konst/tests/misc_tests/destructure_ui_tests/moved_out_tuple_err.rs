type Tuple = (String, String, String);

const fn to_bar(foo: Tuple) -> impl Sized {
    konst::destructure!{(bar, baz, qux) = foo}

    std::mem::forget(foo);

    (bar, baz, qux)
}


fn main(){}