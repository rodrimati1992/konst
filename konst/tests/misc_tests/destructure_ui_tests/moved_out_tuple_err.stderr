error[E0382]: use of moved value: `foo`
 --> tests/misc_tests/destructure_ui_tests/moved_out_tuple_err.rs:6:22
  |
3 | const fn to_bar(foo: Tuple) -> impl Sized {
  |                 --- move occurs because `foo` has type `(String, String, String)`, which does not implement the `Copy` trait
4 |     konst::destructure!{(bar, baz, qux) = foo}
  |     ------------------------------------------ value moved here
5 |
6 |     std::mem::forget(foo);
  |                      ^^^ value used here after move
